Updated the minimal MCP server doc to include updated info for VS, VS… (#51025)
dotnet/docs: docs/ai/quickstarts/build-mcp-server.md

# Create a minimal MCP server using C# and publish to NuGet

In this quickstart, you create a minimal Model Context Protocol (MCP) server using the [C# SDK for MCP](https://github.com/modelcontextprotocol/csharp-sdk), connect to it using GitHub Copilot, and publish it to NuGet (stdio transport only). MCP servers are services that expose capabilities to clients through the Model Context Protocol (MCP).

> [!NOTE]
> The `Microsoft.McpServer.ProjectTemplates` template package is currently in preview.

## Prerequisites

::: zone pivot="visualstudio"

> - [.NET 10.0 SDK](https://dotnet.microsoft.com/download/dotnet) (preview 6 or higher)

Reviewer: I don't think we need to include the required preview version. Surely everyone that's downloaded .NET 10 is using GA at this point?

Author: Fixed!

::: zone-end

::: zone pivot="vscode"

- [.NET 10.0 SDK](https://dotnet.microsoft.com/download/dotnet) (preview 6 or higher)
- [Visual Studio Code](https://code.visualstudio.com/) (optional)
- [C# Dev Kit extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit)
- [GitHub Copilot extension](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot) for Visual Studio Code
- [NuGet.org account](https://www.nuget.org/users/account/LogOn)

::: zone-end

::: zone pivot="cli"

- [.NET 10.0 SDK](https://dotnet.microsoft.com/download/dotnet) (preview 6 or higher)
- [Visual Studio Code](https://code.visualstudio.com/) (optional)
- [Visual Studio](https://visualstudio.microsoft.com/) (optional)
- [GitHub Copilot](https://github.com/features/copilot) / [GitHub Copilot extension](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot) for Visual Studio Code
- [NuGet.org account](https://www.nuget.org/users/account/LogOn)

::: zone-end

## Create the project

::: zone pivot="visualstudio"

1. In a terminal window, install the MCP Server template:

   ```bash
   dotnet new install Microsoft.McpServer.ProjectTemplates
   ```

    > [!NOTE]

>     > .NET 10.0 SDK (Preview 6 or later) is required to install `Microsoft.McpServer.ProjectTemplates`.

Reviewer: ```suggestion
    > .NET 10.0 SDK or a later version is required to install `Microsoft.McpServer.ProjectTemplates`.
```

> 1. Open Visual Studio, and select **Create a new project** in the start window (or select **File > New > Project/Solution..** from inside Visual Studio).

Reviewer: ```suggestion
1. Open Visual Studio, and select **Create a new project** in the start window (or select **File** > **New** > **Project/Solution** from inside Visual Studio).
```

1. Update the `<PackageId>` in the `.csproj` file to be unique on NuGet.org, for example `<NuGet.org username>.SampleMcpServer`.

::: zone-end

## Tour the MCP Server Project

Creating your MCP server project via the template give you the following major files: 

* `Program.cs`: A file defining the application as an MCP server and registering MCP services such as transport type and MCP tools.  

>     * Choosing the (default) **stdio** transport option in when creating the project, this file will be configured to define the MCP Server as a local one (i.e. `.withStdioServerTransport()`).  

Reviewer: ```suggestion
    * If you choose the (default) **stdio** transport option when creating the project, this file will be configured to define the MCP Server as a local one (that is, `.withStdioServerTransport()`).  
```

>     * Choosing the **http** transport option will configure this file to include remote transport-specific definitions (i.e. `.withHttpServerTransport()`, `MapMcp()`). 

Reviewer: ```suggestion
    * If you choose the **http** transport option, this file includes remote transport-specific definitions (that is, `.withHttpServerTransport()`, `MapMcp()`). 
```
